frontend/public/components/graphs/prometheus-graph.tsx

```diff
@@ -19,23 +25,53 @@ export const getPrometheusExpressionBrowserURL = (urls, queries): string => {
   return `${base}/graph?${params.toString()}`;
 };
 
-const getQueryBrowserURL = (query: string): string => {
-  const params = new URLSearchParams();
-  params.set('query0', query);
-  return `/monitoring/query-browser?${params.toString()}`;
-};
+const mapStateToProps = (state: RootState) => ({
+  activePerspective: getActivePerspective(state),
+  canAccessMonitoring:
+    !!state[featureReducerName].get(FLAGS.CAN_GET_NS) && !!window.SERVER_FLAGS.prometheusBaseURL,
+  namespace: getActiveNamespace(state),
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  setActivePerspective: (id: string) => dispatch(UIActions.setActivePerspective(id)),
+});
 
-export const PrometheusGraphLink = ({
+export const PrometheusGraphLink_: React.FC<PrometheusGraphLinkProps> = ({
+  activePerspective,
+  canAccessMonitoring,
   children,
+  namespace,
   query,
-}: React.PropsWithChildren<PrometheusGraphLinkProps>) =>
-  query ? (
-    <Link to={getQueryBrowserURL(query)} style={{ color: 'inherit', textDecoration: 'none' }}>
+  setActivePerspective,
+}) => {
+  const onClick = React.useCallback(() => {
+    if (!canAccessMonitoring && activePerspective !== 'dev') {
```

Comment:
It's unfortunate that the core console needs to know about the dev perspective. This would break if we ever wanted a build that disabled the dev console plugin.